src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -79,6 +79,53 @@ private static async ValueTask<(bool, T?)> TryResolveBodyAsync<T>(HttpContext ht
         }
 """;
 
+    public static string TryResolveFormAsyncMethod => """
```

Comment:
How are we gonna keep these in sync with RDF?

Reply:
This is probaby true for a lot of the things that we've had to duplicate in RDG because we can't do shared source the way we like. I'm wondering whether we can come up with a shared source approach that would work with RDG where we embed certain shared source source files into the RDG assembly and then emit them verbatim into the generated code.

Reply:
Exposing public API 😄

Reply:
> Exposing public API is another way to share code 😄

Yeah, this came up recently in a discussion around the `ExecuteObjectResult` method which is also shared in both places.

Public API is an option but we have to be judicious about where we put things. At this point, we might want to have some sort of `Microsoft.AspNetCore.Http.Generated.Internal` namespace to track all of these shared functions.

Another thing to note is that styles have diverged a little bit and we have to reconcile them. For example, RDF Uses a static Log class and resolves the logger per invocation whereas RDG doesn't.

Anyway, the `ResolveForm` method is interesting because it's more likely to change as we evolve the form implementation so there's going to be bigger requirements to sync between the two.